### PR TITLE
Rem 541

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
new page : 
iv-item-groups

### CODE REVIEW

File: src/pages/iv-item-groups/form/IvItemGroupsForm.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.Group.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
```
Comment (omarZrayka23) [inline src/pages/iv-item-groups/form/IvItemGroupsForm.js:30]: u do not need || null

File: src/pages/iv-item-groups/form/IvItemGroupsForm.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' }
+                ]}
+                values={formik.values}
+                valueField='recordId'
+                displayField={['reference', 'name']}
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('msId', newValue ? newValue.recordId : '')
```
Comment (omarZrayka23) [inline src/pages/iv-item-groups/form/IvItemGroupsForm.js:159]: u can use || instead ? :

File: src/pages/iv-item-groups/form/IvItemGroupsForm.js (hunk 1 above)
Comment (omarZrayka23) [inline src/pages/iv-item-groups/form/IvItemGroupsForm.js:30]: remove || null

File: src/pages/iv-item-groups/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    try {
+      const response = await getRequest({
+        extension: InventoryRepository.Group.qry,
```
Comment (omarZrayka23) [inline src/pages/iv-item-groups/index.js:26]: why .qry and api pagination

**Comment by omarZrayka23** [comment]
Screenshot 2024-09-04 at 9 14 46 AM

after click save and clear

**Comment by Ghinwa-Arab** [comment]
image

missing search, templates combo & buttons

**Comment by Ghinwa-Arab** [comment]
image

Try to add new record then click save and clear, this error appears

**Comment by Ghinwa-Arab** [comment]
image

1- Try to add new record then click save and clear, this error appears 
2- Form shouldn't clear if any error appears while saving

File: src/pages/iv-item-groups/index.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    try {
+      const response = await getRequest({
+        extension: InventoryRepository.Group.page,
+        parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_filter=&_sortField=`
```
Comment (Ghinwa-Arab) [inline src/pages/iv-item-groups/index.js:27]: are u sure it's necessary to send `&_sortField=`  ??

File: src/pages/iv-item-groups/form/IvItemGroupsForm.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { useForm } from 'src/hooks/form'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { ControlContext } from 'src/providers/ControlContext'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
```
Comment (Ghinwa-Arab) [inline src/pages/iv-item-groups/form/IvItemGroupsForm.js:14]: remove, not used

File: src/pages/iv-item-groups/form/IvItemGroupsForm.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                datasetId={DataSets.PROCUREMENT_METHOD}
+                name='procurementMethod'
+                label={labels.procurement}
+                valueField='key'
+                displayField='value'
+                values={formik.values}
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('procurementMethod', newValue?.key || '')
+                  formik.setFieldValue('procurementName', newValue?.value || '')
```
Comment (Ghinwa-Arab) [inline src/pages/iv-item-groups/form/IvItemGroupsForm.js:123]: it's necessary to update procurementName also?

File: src/repositories/InventoryRepository.js [hunk 7]
```diff
@@ -6,7 +6,12 @@ export const InventoryRepository = {
     get: service + 'getIT'
   },
   Group: {
-    qry: service + 'qryGRP'
+    qry: service + 'qryGRP',
```
Comment (Ghinwa-Arab) [inline src/repositories/InventoryRepository.js:9]: remove qry if not used

File: src/pages/iv-item-groups/index.js [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+          extension: InventoryRepository.Group.snapshot,
+          parameters: `_filter=${filters.qry}`
+        })
+      : await fetchGridData(pagination)
+  }
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    const defaultParams = `_startAt=${_startAt}&_pageSize=${_pageSize}`
```
Comment (Ghinwa-Arab) [inline src/pages/iv-item-groups/index.js:59]: refactor this

File: src/repositories/InventoryRepository.js (hunk 7 above)
Comment (Alinadimmokdad) [reply src/repositories/InventoryRepository.js:9]: its used in mapping

**Review by Ghinwa-Arab — APPROVED** [review]

**Pull request merged**